punishments.js

```diff
@@ -42,6 +42,7 @@ const AUTOLOCK_POINT_THRESHOLD = 8;
 /**
  * @augments {Map<string, Punishment>}
  */
+// @ts-ignore TypeScript bug
```

Comment:
How sure are you that this is actually a TypeScript bug?

Reply:
I just copied the comment from https://github.com/Zarel/Pokemon-Showdown/blob/5094631fcf780af9d203c59a4ce78c5ae1cec167/monitor.js#L22 since thats the same error as this

Reply:
I actually do remember having trouble with this, but I'd rather not deem it a TypeScript bug without taking another look. I'd prefer if you marked all the things you have trouble with as `@ts-ignore TODO: possible TypeScript bug` for me to take a look at.